pinc/Project.inc

```diff
@@ -768,14 +769,30 @@ function does_project_page_table_exist($projectid)
 // long form (projectID####) or just the project number (####).
 // If it does, it returns it, otherwise dies with a warning.
 function validate_projectID($param_name, $value, $allownull = false)
+{
+    try
+    {
+        return ex_validate_projectID($param_name, $value, $allownull);
+    }
+    catch(InvalidProjectIDException $exception)
+    {
+        die($exception->getMessage());
+    }
+}
+
+// Checks if $value has the form of a valid project ID, either in its
+// long form (projectID####) or just the project number (####).
+// If it does, it returns it, otherwise throws an exception.
+// this is intended to replace validate_projectID()
+function ex_validate_projectID($param_name, $value, $allownull = false)
```

Comment:
You don't need a new function, just change the die() to throw an exception and let the exception handler handle printing it out.